kernel/src/engine/arrow_expression.rs

```diff
@@ -30,6 +32,18 @@ use crate::schema::{DataType, PrimitiveType, SchemaRef};
 use crate::{EngineData, ExpressionEvaluator, ExpressionHandler};
 
 // TODO leverage scalars / Datum
+//
+pub fn expression_to_row_filter(predicate: Expression) -> RowFilter {
```

Comment:
can it just take `&Expression`?

Reply:
[ArrowPredicateFn](https://docs.rs/parquet/latest/parquet/arrow/arrow_reader/struct.ArrowPredicateFn.html) takes a `FnMut` with a `'static` lifetime requirement. So I think this means only static lifetime references may go in. 

One possible solution may be `Box` or `Arc`ing this, but `expression_to_row_filter` is called once in `try_create_from_parquet`. Unless we're calling `read_parquet_files` multiple times, I don't think it's worth wrapping Expression.

Reply:
Sounds good! 

Regarding `ArrowPredicate`, I briefly considered the it, but its [trait requirements](https://arrow.apache.org/rust/parquet/arrow/arrow_reader/filter/trait.ArrowPredicate.html) are still `'static`. This would [disallow](https://doc.rust-lang.org/rust-by-example/scope/lifetime/static_lifetime.html) *any* non-static lifetimes in a struct that implements the trait. That would include fields.

Reply:
This actually isn't true. It's a common misconception. So much so there's a page about it :) https://github.com/pretzelhammer/rust-blog/blob/master/posts/common-rust-lifetime-misconceptions.md#2-if-t-static-then-t-must-be-valid-for-the-entire-program

Really arrow is just asserting that it can "own" the predicate here.

Reply:
But also I still think it's okay to clone the expression

Reply:
Woah! TIL, thank you